vcd/resource_vcd_vapp.go

```diff
@@ -356,28 +339,25 @@ func resourceVcdVAppUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 
 		if d.HasChange("memory") {
-			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-				task, err := vapp.ChangeMemorySize(d.Get("memory").(int))
-				if err != nil {
-					return resource.RetryableError(fmt.Errorf("error changing memory size: %#v", err))
-				}
 
-				return resource.RetryableError(task.WaitTaskCompletion())
-			})
+			task, err := vapp.ChangeMemorySize(d.Get("memory").(int))
+			if err != nil {
+				return fmt.Errorf("error changing memory size: %#v", err)
+			}
+
+			err = task.WaitTaskCompletion()
 			if err != nil {
 				return err
 			}
 		}
 
 		if d.HasChange("cpus") {
-			err = retryCall(vcdClient.MaxRetryTimeout, func() *resource.RetryError {
-				task, err := vapp.ChangeCPUCount(d.Get("cpus").(int))
-				if err != nil {
-					return resource.RetryableError(fmt.Errorf("error changing cpu count: %#v", err))
-				}
+			task, err := vapp.ChangeCPUCount(d.Get("cpus").(int))
+			if err != nil {
+				fmt.Errorf("error changing cpu count: %#v", err)
```

Comment:
There should be a `return` statement here

Reply:
fixed